components/sup/src/main.rs

```diff
@@ -851,26 +852,39 @@ fn sub_status(m: &ArgMatches) -> Result<()> {
                     process::exit(2);
                 }
             };
-
-            for spec in specs {
-                let status = Manager::service_status(&cfg, &spec.ident)?;
-                outputln!("{}", status);
-            }
+            print_statuses(specs.iter()
+                                .filter_map(|spec| Manager::service_status(&cfg, &spec.ident).ok())
+                                .collect::<Vec<ServiceStatus>>())?;
         }
         None => {
-            let statuses = Manager::status(&cfg)?;
-            if statuses.is_empty() {
-                println!("No services loaded.");
-                return Ok(());
-            }
-            for status in statuses {
-                println!("{}", status);
-            }
+            print_statuses(Manager::status(&cfg)?)?;
         }
     }
     Ok(())
 }
 
+fn print_statuses(statuses: Vec<ServiceStatus>) -> Result<()> {
+    if statuses.is_empty() {
+        println!("No services loaded.");
+        return Ok(());
+    }
+    let mut tw = cli::TableWriter::new(7);
+    tw.set_titles(vec!("package", "type", "state", "uptime", "pid", "group", "style"));
```

Comment:
I'd declare the header vector first, then use its length to create the `TableWriter`, instead of having the magic number `7` in the code.

Reply:
Yep this feels better indeed.